lessons/basics/comprehensions.md

```diff
@@ -105,7 +105,7 @@ iex> for {k, v} <- [one: 1, two: 2, three: 3], into: %{}, do: {k, v}
 %{one: 1, three: 3, two: 2}
 ```
 
-Since bitstrings are enumerable we can use list comprehensions and `:into` to create strings:
+Since binaries are collectable we can use list comprehensions and `:into` to create strings:
```

Comment:
Collectable and Enumable are different. It's quite tricky because you need to look at the source code (and this could be spread out in separate files) to see what module implements what protocol.
The documentation for Collectable [explains this here](https://hexdocs.pm/elixir/Collectable.html#module-why-collectable):

Most of collectables are enumerable, but BitString is not.
you can see this in : 
```elixir
iex(1)> "string" |> Enum.to_list
** (Protocol.UndefinedError) protocol Enumerable not implemented for "string"
    (elixir) lib/enum.ex:1: Enumerable.impl_for!/1
    ...
```

I have scanned the docs to see what's implemente in Enumerable and in Collectable.

__Collectable__
- BitString
- File.Stream
- HashDict
- HashSet
- IO.Stream
- List
- Map
- MapSet

__Enumerable__
- File.Stream
- Function
- GenEvent.Stream
- HashDict
- HashSet
- IO.Stream
- List
- Map
- MapSet
- Range
- Stream

For those interested in the commands I used, I searched the source code.
I could be getting results from examples. But I feel they are quite accurate.

https://gist.github.com/eksperimental/701103fc9e723fddc0bfeb757ecb1e7d